### DISCUSSION TITLE
MediaPlayer for Avalonia

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#7198] [General]
[I've ported my MediaPlayer UI to support Avalonia](https://github.com/mysteryx93/MediaPlayerUI.NET)

Currently it only supports audio playback through BASS, cross-platform with optional pitch/rate/speed control.

In the future, I want to support MPV media player (need to implement OpenGL playback, and fullscreen UI), but it may take some time before I get to this, and if someone wants to beat me to it, go ahead!

Also the styles aren't 100% functional in Avalonia. I couldn't find a way to get the geometry transparency for the volume control, and I didn't do the speed slider yet.

Contributions wanted for additional styles and color themes! I've set up the project structure so that we can select the visual theme and color theme separately; and possibly combine them as we want.

![Screenshot](https://github.com/mysteryx93/MediaPlayerUI.NET/blob/master/Screenshot.png?raw=true)

**Comment by kuiperzone** [comment]
Good stuff!

**Comment by Tammon23** [comment]
Lovely work done here!

Is there any update towards adding/working on video support for this project as of 2023?

**Reply by mysteryx93** [reply]
Someone made an attempt; with still some issues. If someone can work out the OpenGL integration API in a way that works properly cross-platform, I can do the integration with MediaPlayerUI and add full-screen mode and such.
https://github.com/mysteryx93/MediaPlayerUI.NET/issues/5

**Reply by Tammon23** [reply]
Thanks for the link, I'll check that out.

**Comment by ivanjx** [comment]
does your library support using available media player on the os (avfoundation on ios, exoplayer on android, etc) ?

**Reply by mysteryx93** [reply]
Are those raw video playback features without any UI?

You can bind this UI to those renderers by creating your own implementation; and can submit your implementations with a pull request.

Also if you want to customize the UI layout for mobile devices, that's also rather easy to do.
